hi-ml-azure/src/health_azure/himl.py

```diff
@@ -685,67 +648,72 @@ def _str_to_path(s: Optional[PathOrString]) -> Optional[Path]:
-        # InvalidInput { message: "invalid uri format"
-        data_path = data_asset.path
-
-        inputs[input_name] = Input(
-            # Data assets can be of type "uri_folder", "uri_file", "mltable", all of which are value types in Input
-            type=data_asset.type,  # type: ignore
-            path=data_path,
-            mode=InputOutputModes.MOUNT if input_dataset.use_mounting else InputOutputModes.DOWNLOAD
-        )
-    return inputs
+
+        data_assets.append(data_asset)
+
+    return data_assets
+
+
+def create_v2_inputs(ml_client: MLClient, input_datasets: List[DatasetConfig]) -> Dict[str, Input]:
+    """
+    Create a dictionary of Azure ML v2 Input objects, required for passing input data in to an AML job
+
+    :param ml_client: An MLClient object.
+    :param input_datasets: A list of DatasetConfigs to convert to Inputs.
+    :return: A dictionary in the format "input_name": Input.
+    """
+    input_assets = get_data_assets_from_configs(ml_client, input_datasets)
+    # Data assets can be of type "uri_folder", "uri_file", "mltable", all of which are value types in Input
+    return {
+        f"INPUT_{i}": Input(  # type: ignore
+            type=data_asset.type,
+            path=data_asset.path,
+            mode=InputOutputModes.MOUNT if input_datasets[i].use_mounting else InputOutputModes.DOWNLOAD
+        ) for i, data_asset in enumerate(input_assets)
+    }
 
 
-def create_v2_outputs(output_datasets: List[DatasetConfig]) -> Dict[str, Output]:
+def create_v2_outputs(ml_client: MLClient, output_datasets: List[DatasetConfig]) -> Dict[str, Output]:
     """
     Create a dictionary of Azure ML v2 Output objects, required for passing output data in to an AML job
 
     :param output_datasets: A list of DatasetConfigs to convert to Outputs.
     :return: A dictionary in the format "output_name": Output.
     """
-    outputs = {}
-    for i, output_dataset in enumerate(output_datasets):
-        output_name = f"OUTPUT_{i}"
-        v1_datastore_path = f"azureml://datastores/{output_dataset.datastore}/paths/{output_dataset.name}"
-        # Note that there are alternative formats that the output path can take, such as:
-        # v2_data_asset_path = f"azureml:{output_dataset.name}@latest"
-        outputs[output_name] = Output(  # type: ignore
-            type=AssetTypes.URI_FOLDER,
-            path=v1_datastore_path,
-            mode=InputOutputModes.DIRECT,
-        )
-    return outputs
+
+    output_assets = get_data_assets_from_configs(ml_client, output_datasets)
+    return {
+        # Data assets can be of type "uri_folder", "uri_file", "mltable", all of which are value types in Input
+        f"OUTPUT_{i}": Output(  # type: ignore
+            type=data_asset.type,
+            path=data_asset.path,
+            mode=InputOutputModes.MOUNT,  # hard-coded to mount for now, as this is the only mode that doesn't break
```

Comment:
Direct caused very strange errors where the job would fail and also give 404s if I tried to read the logs of the failed job. Planning on raising an issue with AzureML on this one! There is also the option of `UPLOAD` but that failed silently / isn't the right mode for the job, as it would claim the files were uploaded but I could not see them.